src/utilities/time.ts

```diff
@@ -47,23 +206,167 @@ function parseDurationString(durationString: string): ParsedDurationString | nev
     };
   }
 
-  const fullMicrosecondsRegex = new RegExp(`^${validDurationValueRegex}$`);
+  const isEpoch = validateTime(durationString, TimeTypes.EPOCH);
+  matches = isEpoch ? EPOCH_TIME.exec(durationString) : RELATIVE_TIME.exec(durationString);
+  if (matches !== null) {
+    const [, sign = '', days = '0', hours = '0', minutes = '0', seconds = '0', milliseconds = '0'] =
+      matches as RegExpExecArray;
+
+    const hoursNum = Number(hours);
+    const minuteNum = Number(minutes);
+    const secondsNum = Number(seconds);
+    const millisecondNum = Number(milliseconds);
 
-  if (fullMicrosecondsRegex.test(durationString)) {
-    const microseconds = parseFloat(durationString);
     return {
-      days: 0,
-      hours: 0,
-      isNegative: microseconds < 0,
-      microseconds,
-      milliseconds: 0,
-      minutes: 0,
-      seconds: 0,
+      days: days !== undefined ? Number(days.replace('T', '')) : 0,
```

Comment:
I think that the regex itself should exclude the "T" from the result so that we don't have to always remember to remove it when using the matches.

Reply:
Made updates to the regex to ignore them